### DISCUSSION TITLE
How to generate a summary of misconfiguration similar to tfsec output?

### DISCUSSION THREAD

**Discussion by lhoupert** [aquasecurity/trivy#5924] [Q&A] [answered]
### Question

Hi, when running 

```
docker run --rm -it -v "$(pwd):/src" aquasec/trivy config --report all /src
```
 I don't obtain a summary of the scan.

Is there a flag that can do that? I would like to see a summary similar to the one below which is obtained by running the tfsec image:
```
docker run --rm -it -v "$(pwd):/src" aquasec/tfsec /src 
```

```

  timings
  ──────────────────────────────────────────
  disk i/o             53.939701ms
  parsing              739.510216ms
  adaptation           66.645258ms
  checks               8.878186ms
  total                868.973361ms

  counts
  ──────────────────────────────────────────
  modules downloaded   0
  modules processed    45
  blocks processed     1292
  files read           193

  results
  ──────────────────────────────────────────
  passed               184
  ignored              543
  critical             0
  high                 0
  medium               0
  low                  24

  184 passed, 543 ignored, 24 potential problem(s) detected.
```

Thank you

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

**Comment by nikpivkin** [comment]
@lhoupert  Trivy doesn't output the same summary as tfsec. What are you missing from the report? Metrics?

**Comment by lhoupert** [comment]
Yes I am missing an "agglomerated" summary. 
```
 Metrics
  ──────────────────────────────────────────
  SUCCESSES               184
  EXCEPTIONS              543
  FAILURES              24
  CRITICAL             0
  HIGH                 0
  MEDIUM               0
  LOW                  24
  UNKNOWN        0
```

**Reply by simar7** [reply]
We have the summary as such:

```shell
trivy config <input>

main.tf (terraform)

Tests: 14 (SUCCESSES: 7, FAILURES: 5, EXCEPTIONS: 2)
Failures: 5 (UNKNOWN: 0, LOW: 2, MEDIUM: 2, HIGH: 1, CRITICAL: 0)
```


You can also write your own custom template to format output and pass it to Trivy with the `--template` flag.

**Reply by lhoupert** [reply]
Hi @simar7 this would be good too. The only is I would like to have a summary agglomerating all the the tests and failures of the scan of the individual files for a specific folder (Basically the sum of all tests and failures). Can the custom template do that? If yes, do you have an example I could get some inspirations from?

**Reply by simar7** [reply, marked as answer]
You can read the docs here:  https://aquasecurity.github.io/trivy/v0.48/docs/configuration/reporting/#custom-template

Furthermore, you can take a look at sending output to your custom plugin https://aquasecurity.github.io/trivy/v0.48/docs/configuration/reporting/#plugin
